service/request.go

```diff
@@ -49,8 +51,31 @@ func (r *Request) PoolID(w *ResponseWriter) (ksuid.KSUID, bool) {
 	return r.TagFromPath("pool", w)
 }
 
-func (r *Request) BranchID(w *ResponseWriter) (ksuid.KSUID, bool) {
-	return r.TagFromPath("branch", w)
+func (r *Request) CommitID(w *ResponseWriter) (ksuid.KSUID, bool) {
+	return r.TagFromPath("commit", w)
+}
+
+func (r *Request) decodeCommitMessage(w *ResponseWriter) (api.CommitMessage, bool) {
+	commitJSON := r.Header.Get("Zed-Commit")
+	var message api.CommitMessage
+	if commitJSON != "" {
+		if err := json.Unmarshal([]byte(commitJSON), &message); err != nil {
+			w.Error(fmt.Errorf("load endpoint encountered invalid JSON in Zed-Commit header: %w", err))
+			return message, false
+		}
+	}
+	return message, true
+}
+
+func (r *Request) StringArg(w *ResponseWriter, arg string) (string, bool) {
```

Comment:
Nit: Give this a name and signature in keeping with the other methods here.
```suggestion
func (r *Request) StringFromPath(arg string, w *ResponseWriter) (string, bool) {
```